odoo/addons/test_http/tests/test_webjson2.py

```diff
@@ -216,6 +236,32 @@ def test_webjson2_api_model(self):
         self.assertEqual(res.status_code, HTTPStatus.UNPROCESSABLE_ENTITY)
         self.assertEqual(res.headers.get('Content-Type'), 'application/json; charset=utf-8')
 
+    def test_webjson2_binary_fields(self):
+        # reading an image on the user
+        res = self.db_url_open(
+            '/json/2/res.users/read',
+            data=r'{"ids": [%d], "fields": ["image_128"]}' % self.jackoneill.id,
+            headers=CT_JSON | self.bearer_header,
+        )
+        self.assertIn('"image_128": "PD94bWw', res.text, "Base64 encoded SVG missing")  # starts with b'<?xml version...'
+        self.assertEqual(res.status_code, HTTPStatus.OK)
+        self.assertEqual(res.headers.get('Content-Type'), 'application/json; charset=utf-8')
+
+        attachment = self.env['ir.attachment'].with_user(self.jackoneill).create({
+            'name': 'test',
+            'raw': b'x  \x01-',
+        })
+        raw_dump = json.dumps(attachment.raw, ensure_ascii=False, default=json_default)
+        values = (attachment.id, attachment.checksum, attachment.datas.decode(), raw_dump.strip('"'))
+        res = self.db_url_open(
+            '/json/2/ir.attachment/search_read',
+            data=r'{"domain": [["id", "=", %d]], "fields": ["checksum", "datas", "raw"], "limit": 1}' % attachment.id,
+            headers=CT_JSON | self.bearer_header,
+        )
+        self.assertEqual(res.text, '[{"id": %d, "checksum": "%s", "datas": "%s", "raw": "%s"}]' % values)
```

Comment:
I don't think it is worth to read `checksum`. And I would like to read the actual string/bytes literal value for datas and raw. Not `attachment.datas.decode()` or `json.dumps(attachment.raw, ensure_ascii=False, default=json_default).strip('"')`